pkg/controllers/chart/chart.go

```diff
@@ -196,6 +197,10 @@ func (c *Controller) OnRemove(key string, chart *helmv1.HelmChart) (*helmv1.Helm
 		return nil, err
 	}
 
+	// sleep for 3 seconds to give the job time to perform the helm install
+	// before emitting any errors
+	time.Sleep(3 * time.Second)
```

Comment:
Do we need to sleep here, or can we just re-enqueue the current object for processing after a few seconds?

Reply:
We could just remove it, but it will generate significantly more noisy logs since we're going to re-enqueue it again and again while it is still waiting for deletion. We don't have to re-enqueue cause an error will be thrown in the code below if we just let it progress

https://github.com/aiyengar2/helm-controller/blob/8bb938728c76b0ba70aee7f611e87f84afe31420/pkg/controllers/chart/chart.go#L240-L249

Reply:
However, we can't get around returning `nil` instead of an error here.

The moment we return a nil error, the OnRemove handler will proceed to remove the finalizer from the object: https://github.com/rancher/wrangler/blob/b54fe3a4ff86983c072cfe6d7bdf1148757c2e28/pkg/generic/remove.go#L47-L56

As a result, the next time the object is enqueued, it should already be deleted from the cache / store, so the enqueue operation will happen with a nil object: https://github.com/rancher/lasso/blob/749b3397db38d63da5dfdb4f80ac1c9a0dcdfaa8/pkg/controller/controller.go#L229-L231

This would result in doing a noop on a re-enqueue: https://github.com/aiyengar2/helm-controller/blob/8bb938728c76b0ba70aee7f611e87f84afe31420/pkg/controllers/chart/chart.go#L199-L201